iceoryx_posh/test/moduletests/test_popo_server_port_user.cpp

```diff
@@ -0,0 +1,732 @@
+{
+    ::testing::Test::RecordProperty("TEST_ID", "29b0a347-2439-488d-ba13-962c7acffac1");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+
+    EXPECT_FALSE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithOneRequestButIntermediatelyHavingNoneIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a9fc0d89-7b97-48c4-8ba4-e4b22355116b");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithMultipleRequestsIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "30e36e69-4f23-41ee-aac7-d34240f075ae");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{2U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithMultipleRequestsAndAllButOneRemovedIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "96c42094-9931-46a3-8fd6-e7d28b490527");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{2U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithNoRequestsButIntermediatellyHavingMultipleRequestsIsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "44845e16-ec0a-4b4e-a000-ff62f377c0b9");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{2U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+
+    EXPECT_FALSE(sut.portUser.hasNewRequests());
+}
+
+// END hasNewRequests tests
+
+// BEGIN getRequest tests
+
+TEST_F(ServerPort_test, GetRequestWithoutOfferResultsIn_NO_CHUNK_AVAILABLE)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6c555224-8b03-46c5-a988-059ac2656149");
+    auto& sut = serverPortWithoutOfferOnCreate;
+
+    sut.portUser.getRequest()
+        .and_then([&](const auto&) { FAIL() << "Expected ChunkReceiveResult::NO_CHUNK_AVAILABLE but got request"; })
+        .or_else([&](const auto& error) { EXPECT_THAT(error, Eq(ChunkReceiveResult::NO_CHUNK_AVAILABLE)); });
+}
+
+TEST_F(ServerPort_test, GetRequestWithNoRequestsResultsIn_NO_CHUNK_AVAILABLE)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2b78536b-6902-4e55-aef2-654da5afdb80");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    sut.portUser.getRequest()
```

Comment:
Here I would add another value to `ServerPortRequestResult` `NO_REQUESTS_PENDING`.

Reply:
done